audiobook_generator/tts_providers/openai_tts_provider.py

```diff
@@ -0,0 +1,91 @@
+from audiobook_generator.tts_providers.base_tts_provider import BaseTTSProvider
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_supported_models():
+    return ["tts-1", "tts-1-hd"]
+
+
+def get_supported_voices():
+    return ["alloy", "echo", "fable", "onyx", "nova", "shimmer"]
+
+
+def get_supported_formats():
+    return ["mp3", "acc", "flac"]
+
+
+class OpenAITTSProvider(BaseTTSProvider):
+    def __init__(self, config: GeneralConfig):
+        super().__init__(config)
+        self.model = config.model_name or "tts-1"
+        self.voice = config.voice_name or "alloy"
+        self.format = config.output_format or "mp3"
+
+        self.price = 0.03 if self.model == "tts-1-hd" else 0.015
+        # per 1000 characters (0.03$ for HD model, 0.015$ for standard model)
+
+        self.client = OpenAI()  # User should set OPENAI_API_KEY environment variable
+
+    def __str__(self) -> str:
+        return super().__str__()
+
+    def text_to_speech(self, text: str, output_file: str, audio_tags: AudioTags):
+        max_chars = 4000  # should be less than 4096 for OpenAI
+
+        text_chunks = split_text(text, max_chars, self.config.language)
+
+        audio_segments = []
+
+        for i, chunk in enumerate(text_chunks, 1):
+            logger.debug(
+                f"Processing chunk {i} of {len(text_chunks)}, length={len(chunk)}, text=[{chunk}]"
+            )
+            logger.info(
+                f"Processing chapter-{audio_tags.idx} <{audio_tags.title}>, chunk {i} of {len(text_chunks)}"
+            )
+
+            logger.debug(f"Text: [{chunk}], length={len(chunk)}")
+
+            # NO retry for OpenAI TTS because SDK has built-in retry logic
+            response = self.client.audio.speech.create(
+                model=self.model,
+                voice=self.voice,
+                input=chunk,
+                response_format=self.format,
+            )
+            audio_segments.append(io.BytesIO(response.content))
+
+        with open(output_file, "wb", encoding='utf-8') as outfile:
+            for segment in audio_segments:
+                segment.seek(0)
+                outfile.write(segment.read())
+
+        set_audio_tags(output_file, audio_tags)
+
+    def get_break_string(self):
+        return "   "
+
+    def get_output_file_extension(self):
+        return self.config.output_format
+
+    def validate_config(self):
+        if self.config.model_name not in get_supported_models():
+            raise ValueError(f"OpenAI: Unsupported model name: {self.config.model_name}")
+        if self.config.voice_name not in get_supported_voices():
+            raise ValueError(f"OpenAI: Unsupported voice name: {self.config.voice_name}")
+        if self.config.output_format not in get_supported_formats():
+            raise ValueError(f"OpenAI: Unsupported output format: {self.config.output_format}")
+
+    def estimate_cost(self, total_chars):
+        return math.ceil(total_chars / 1000) * self.price
```

Comment:
Same thing here. `math.ceil(total_chars / 1000  * self.price)`

Reply:
no, no, that's correct